### PR TITLE
Remove writers from TypedRecordProcessor

### LINKED ISSUE #8002: Remove writes from TypedRecordProcessor interface
**Description**

On the migration to event sourcing, we replaced the usage of `TypedStreamWriter` and `TypedResponseWriter` from `TypedRecordProcessor` with the `Writes` object. We should remove them from the interface to avoid unintended usage.

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

 * Replaces the usage of the StreamWriter and ResponseWriter from the parameters with the Writers and their "sub-writers"
   * Discussable whether we always want to have Writers in the processor classes or just the specific writers, but I guess we can here improve also later
 * Adjust several test cases that they also are filled with the Writers object
 * Finally, remove the writers from the processRecord interface

## Related issues

related to https://github.com/camunda/zeebe/issues/9727
closes https://github.com/camunda/zeebe/issues/8002

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/TypedRecordProcessor.java [hunk 1]
```diff
@@ -23,32 +22,16 @@
   default void processRecord(final TypedRecord<T> record) {}
 
   /**
-   * @see #processRecord(TypedRecord, LegacyTypedResponseWriter, LegacyTypedStreamWriter, Consumer)
+   * @see #processRecord(TypedRecord, Consumer)
```
Comment (pihme) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/TypedRecordProcessor.java:25]: as far as I can tell, this also closes #8002 So the TODO at the top can be removed

Comment (Zelldon) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/TypedRecordProcessor.java:25]: Good point thanks 😊

File: engine/src/main/java/io/camunda/zeebe/engine/processing/ProcessEventProcessors.java [hunk 2]
```diff
@@ -67,7 +66,8 @@ public static TypedRecordProcessor<ProcessInstanceRecord> addProcessProcessors(
 
     final var processEngineMetrics = new ProcessEngineMetrics(zeebeState.getPartitionId());
 
-    addProcessInstanceCommandProcessor(typedRecordProcessors, zeebeState.getElementInstanceState());
+    addProcessInstanceCommandProcessor(
```
Comment (pihme) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/ProcessEventProcessors.java:69]: :x: In this class (but not in any of the changed lines) `streamWriter` and `responseWriter` are no longer used. Removing them here allows to remove them from `EngineContext` as well.

File: engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/TypedRecordProcessor.java [hunk 3]
```diff
@@ -57,9 +40,7 @@ default void processRecord(
   default void processRecord(
       final long position,
       final TypedRecord<T> record,
-      final LegacyTypedResponseWriter responseWriter,
-      final LegacyTypedStreamWriter streamWriter,
       final Consumer<SideEffectProducer> sideEffect) {
-    processRecord(record, responseWriter, streamWriter, sideEffect);
+    processRecord(record, sideEffect);
   }
```
Comment (saig0) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/TypedRecordProcessor.java:45]: :wrench: We could remove this method from the interface. It is not implemented by any processor.

**Review by saig0 — APPROVED** [review]
@Zelldon looks good :+1: 

I have just one minor suggestion.

File: engine/src/main/java/io/camunda/zeebe/engine/processing/ProcessEventProcessors.java (hunk 2 above)
Comment (Zelldon) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/ProcessEventProcessors.java:69]: Not sure whether I can completely follow but I think what you want is done via #10000

**Comment by Zelldon** [comment]
Thank you both for your review :) :hugs:

**Pull request merged**